pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -3493,6 +3495,10 @@ private CompletableFuture<Boolean> isAllowAutoTopicCreationAsync(final TopicName
     }
 
     public boolean isDefaultTopicTypePartitioned(final TopicName topicName, final Optional<Policies> policies) {
+        if (topicName.getPartitionedTopicName().endsWith(DLQ_GROUP_TOPIC_SUFFIX)
```

Comment:
I suggest we collect all the topic names like this to be used as a keyword collection in Pulsar. 

Plus, it's a breaking change for existing users using a topic name with `-DLQ` suffix. and this suffix is kinda common. we should give a plan for this breaking change. (workaround or notice important)

Reply:
Good suggestion, we can do this improvement with a separate PIP, just like the discussion https://lists.apache.org/thread/qgbpzr6o3k5rqbs2jvpkdh8hr9jpnw39.

Before that, we can fix the issue faster through this PR.

Reply:
Yes, but what is your plan for breaking changes?  I saw you also wanna cherry-pick it to branch 3.0

Reply:
Because the issue described in the motivation will generate much more dirty partitioned&non-partitioned topics

Reply:
Yes, it's a case. What if the user uses the topic name with the `-DLQ` suffix as a regular topic?

Reply:
> Yes, it's a case. What if the user uses the topic name with the -DLQ suffix as a regular topic?

- They can use the topics that were created before as before.
- For new topics, they should create manually if they want a partitioned topic with suffix `-DLQ`.

Reply:
Okay, Let us add a notice in the new release for this breaking change. thanks! :)

Reply:
make sense.

Reply:
+1, Should we add make `release/importent-notic` label?

Reply:
I added the label, please feel free to remove it if you think it's not suitable